openid-sse-framework-1_0.txt

```diff
@@ -1504,21 +1617,182 @@ Tulshibagwale, et al.   Expires December 10, 2021              [Page 26]
+       policy does not permit the Receiver from obtaining the status,
+       then the Transmitter MAY respond with a 403 error status.
+
+   3.  If the Receiver requests the status for a stream that does not
+       exist then the Transmitter MUST respond with a 404 error status.
+
+   4.  If the Receiver requests the status for specific Subject, but the
+       Transmitter policy does not permit the Receiver to read the
+       status of that Subject, then the Transmitter MAY respond with a
+       404 error status in order to not reveal the policy decision.
+
+   5.  If the specified Subject is invalid then the Transmitter MUST
+       respond with a 404 error status.
+
+7.1.2.2.  Updating a Stream's Status
+
+   An Event Receiver updates the current status of a stream by making an
+   HTTP POST request to the Status Endpoint.  The POST body contains a
+   JSON [RFC7159] object with the following fields:
+
+   stream_id
+      REQUIRED.  The stream whose status is being updated.
+
+   status
+      REQUIRED.  The new status of the Event Stream.
+
+
+
+
+Tulshibagwale, et al.        Standards Track                   [Page 30]
+
+                          openid-sse-framework                 June 2021
+
+
+   subject
+      OPTIONAL.  The Subject to which the new status applies.
+
+   reason
+      OPTIONAL.  A short text description that explains the reason for
+      the change.
+
+   On receiving a valid request the Event Transmitter responds with a
+   200 OK response containing a JSON [RFC7159] representation of the
+   updated stream status in the body.
+
+   The following is a non-normative example request to update an Event
+   Stream's status:
+
+   POST /sse/status HTTP/1.1
+   Host: transmitter.example.com
+   Authorization: Bearer eyJ0b2tlbiI6ImV4YW1wbGUifQo=
+
+   {
+     "stream_id": "f67e39a0a4d34d56b3aa1bc4cff0069f",
+     "status": "paused"
+   }
+
+
+     Figure 27: Example: Update Stream Status Request Without Optional
+                                   Fields
+
+   The following is a non-normative example of an Update Stream Status
+   request with optional fields:
+
+   POST /sse/status HTTP/1.1
+   Host: transmitter.example.com
+   Authorization: Bearer eyJ0b2tlbiI6ImV4YW1wbGUifQo=
+
+   {
+     "stream_id": "f67e39a0a4d34d56b3aa1bc4cff0069f",
+     "status": "paused",
+     "subject": {
+       "tenant" : {
+         "format" : "iss_sub",
+         "iss" : "http://example.com/idp1",
+         "sub" : "1234"
+       }
+     },
+     "reason": "Disabled by administrator action."
+   }
+
+
+
+
+
+Tulshibagwale, et al.        Standards Track                   [Page 31]
+
+                          openid-sse-framework                 June 2021
+
+
+   Figure 28: Example: Update Stream Status Request With Optional Fields
+
+   The following is a non-normative example response:
+
+   HTTP/1.1 200 OK
+   Content-Type: application/json
+   Cache-Control: no-store
+
+   {
+     "status": "paused"
```

Comment:
Should the response contain the stream Id at a minimum, and perhaps the subject for which the status was updated?

Reply:
Good catch!